### ISSUE TITLE
Can't access template's hash parameters' values from #each block within template

### ISSUE THREAD

**Issue by cryarchy** [sunng87/handlebars-rust#416]
I am not sure if this is supported by the handlebars spec or it is achievable in a different way. I am trying to access a variable passed in as a template's hash parameter within an #each block contained by the template. I can access the variable outside of the #each block but cannot seem to find a way to access it within the #each block.

The following code:

```rust
use serde_json;

use handlebars::Handlebars;

fn main() -> Result<(), Box<dyn std::error::Error>> {
  let outer = r#"
    {{>inner name="fruit:" vegetables=fruits}}
  "#;

  let inner = r#"
    {{#each vegetables}}
      {{../name}} {{this}},
    {{/each}}
  "#;

  let mut handlebars = Handlebars::new();
  handlebars.register_template_string("outer", outer)?;
  handlebars.register_template_string("inner", inner)?;
  let data = serde_json::json!({ "fruits": vec!["carrot", "tomato"] });
  assert_eq!(
    handlebars
      .render("outer", &data)?
      .replace(" ", "")
      .replace("\n", ""),
    "fruit:carrot,fruit:tomato".to_owned()
  );
  Ok(())
}
```

fails with:

```
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `"carrot,tomato,"`,
 right: `"fruit:carrot,fruit:tomato"`', src/main.rs:20:3
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Should this be possible? Am I missing something?

**Comment by sunng87** [comment]
Verified [here](https://handlebarsjs.com/playground.html#format=1&currentExample=%7B%22template%22%3A%22%7B%7B%3Einner%20name%3D%5C%22fruit%3A%5C%22%20vegetables%3Dfruits%7D%7D%22%2C%22partials%22%3A%5B%7B%22name%22%3A%22inner%22%2C%22content%22%3A%22%7B%7B%23each%20vegetables%7D%7D%5Cn%20%20%20%7B%7B..%2Fname%7D%7D%20%7B%7Bthis%7D%7D%2C%5Cn%7B%7B%2Feach%7D%7D%22%7D%5D%2C%22input%22%3A%22%7B%20%5C%22fruits%5C%22%3A%20%5B%5C%22carrot%5C%22%2C%20%5C%22tomato%5C%22%5D%20%7D%22%2C%22output%22%3A%22%20%20%20fruit%3A%20carrot%2C%5Cn%20%20%20fruit%3A%20tomato%2C%5Cn%22%2C%22preparationScript%22%3A%22%2F%2F%20Handlebars.registerHelper('loud'%2C%20function(string)%20%7B%5Cn%2F%2F%20%20%20%20return%20string.toUpperCase()%5Cn%2F%2F%20%7D)%3B%5Cn%22%2C%22handlebarsVersion%22%3A%224.7.6%22%7D) that it works on handlebarsjs.

It seems the bug was related with the context of partial. Let me check .

**Comment by sunng87** [comment]
@cryarchy this should be fixed in #419 . I will back port this fix to 3.x branch and create a release later today.

**Comment by cryarchy** [comment]
Awesome. Thank you @sunng87
